docs/docs/development/studio-setup.md

```diff
@@ -0,0 +1,87 @@
+---
+title: Studio Setup
+---
+
+:::warning[Alpha Feature]
+
+ZMK Studio support is in alpha. Although best efforts are being made, backwards compatibility during active development is not guaranteed.
+
+:::
+
+This guide will walk you through enabling ZMK Studio support for a keyboard.
+
+The main additional pieces needed for Studio support involve additional metadata needed in order
+to properly to display the physical layouts available for the particular keyboard.
+
+# Physical Layout Positions
+
+Physical layouts are described as part of the [new shield guide](./new-shield.mdx#physical-layouts) with the exception of the `keys` property that is required for Studio support. This is used to describe the physical attributes of each key position present in that layout and its items are listed in the same order as keymap bindings, matrix transforms, etc. The properties available are:
+
+- Width
+- Height
+- X
+- Y
+- Rotation
+- Rotation X
+- Rotation Y
+
+All X/Y/Width/Height values are in "key unit" units, with a 100th of a key unit resolution. Rotations are in degrees, with a 100th of a degree resolution. Due to the nature of devicetree, all values are stored in the devicetree without decimals as hundredths of a value, e.g. one key unit is represented as `100`, twenty degrees is represented as `2000`, etc.
+
+Width/Height values should be positive and X/Y should be nonnegative.
+If the Rotation value is non-zero, the key will be rotated by that angle around the coordinates specified by Rotation X/Y values, after it is translated to X/Y coordinates. Rotation is clockwise for positive values and counterclockwise otherwise.
```

Comment:
```suggestion
| Property | Type | Description                                             | Unit |
| ---------| ---- | ------------------------------------------ | ------- |
| Width    | int (>0) | Key(cap) width                                    | [centi-](https://en.wikipedia.org/wiki/Centi-)"keyunit"|
| Width    | int (>0) | Key(cap) height                                 | [centi-](https://en.wikipedia.org/wiki/Centi-)"keyunit"|
| X           | uint         | Key X position (center point)            | [centi-](https://en.wikipedia.org/wiki/Centi-)"keyunit"|
| Y           | uint         | Key Y position (center point)            | [centi-](https://en.wikipedia.org/wiki/Centi-)"keyunit"|
| Rotation | int         | Key rotation (clockwise = positive)    | [centi-](https://en.wikipedia.org/wiki/Centi-)degree |
| Rotation X | int         | Rotation origin X position relative to key X position | [centi-](https://en.wikipedia.org/wiki/Centi-)"keyunit"|
| Rotation Y | int         | Rotation origin Y position relative to key Y position | [centi-](https://en.wikipedia.org/wiki/Centi-)"keyunit"|

```
Wrote out a (poorly formatted) example of what I'm thinking.

Reply:
Updated, but with some tweaks for accuracy.

Reply:
The updated version reads to me as if the rotation origin positions are no longer relative to the key. "Rotation origin relative X position"? If they are no longer relative then the 0,0 case needs to be stated explicitly.

Reply:
They are not relative to the X/Y position, but absolute. I can include the term "absolute", if you think that's descriptive enough.